src/app/client/src/app/modules/observation/service/observation-util/observation-util.service.spec.ts

```diff
@@ -1,12 +1,236 @@
+import { ObservationUtilService } from "./observation-util.service";
+import { UserService, KendraService } from "@sunbird/core";
+import { ModalConfig, ModalControls, SuiModal, SuiModalModule, SuiModalService } from "ng2-semantic-ui";
+import { HttpClient } from "@angular/common/http";
+import { HttpClientTestingModule } from "@angular/common/http/testing";
+import { configureTestSuite } from "@sunbird/test-util";
+import { ConfigService, ResourceService, IUserData } from "@sunbird/shared";
+import { CacheService } from "ng2-cache-service";
+import { APP_BASE_HREF } from "@angular/common";
+import { Router } from "@angular/router";
+import {
+  ResponseData,
+  EntityResponse,
+  ProfileDataList,
+  metaData,
+} from "./observation-util.service.spec.data";
+import {
+  of as observableOf,
+  throwError as observableThrowError,
+  Observable,
+  of,
+  observable,
+} from "rxjs";
 
-import { ObservationUtilService } from './observation-util.service';
+describe("ObservationUtilService", () => {
+  let baseHref, kendraService, userService,modalService;
+  let service: ObservationUtilService;
+  let requiredFields, data, allFieldsPresent, originalTimeout;
+  let mockModalRef: SuiModalService;
+  configureTestSuite();
+  beforeEach(() =>
+    TestBed.configureTestingModule({
+      providers: [
+        HttpClient,
+        UserService,
+        KendraService,
+        ConfigService,
+        CacheService,
+        { provide: APP_BASE_HREF, useValue: baseHref },
+        { provide: Router },
+        { provide: ResourceService },
+      ],
+      imports: [HttpClientTestingModule, SuiModalModule],
+    })
+  );
 
-xdescribe('ObservationUtilService', () => {
-  beforeEach(() => TestBed.configureTestingModule({}));
+  beforeEach(() => {
+    service = TestBed.get(ObservationUtilService);
+    userService = TestBed.get(UserService);
+    kendraService = TestBed.get(KendraService);
+    modalService=TestBed.get(SuiModalService);
+    originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
+  });
+
+  afterEach(() => {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
+  });
 
-  it('should be created', () => {
-    const service: ObservationUtilService = TestBed.get(ObservationUtilService);
+  it("should be created", () => {
     expect(service).toBeTruthy();
   });
+
+  it("should run #getProfileData() not null", async () => {
+    userService.userData$ = observableOf({
+      userProfile: {
+        profileUserType: {
+          subType: "deo",
+          type: "administrator",
+        },
+      },
+    });
+    spyOn(service, "getProfileData").and.callThrough();
+    service.getProfileData();
+  });
+  
+
+  it("should run #getProfileData() null", async () => {
+    userService.userData$ = observableOf({
+      userProfile: {
+        profileUserType: {
+          subType: null,
+          type: "administrator",
+        },
+      },
+    });
+    spyOn(service, "getProfileData").and.callThrough();
```

Comment:
these test cases are not proper @pradeepkumarcm96  you need to check something inside test cases,please review all test cases